### DISCUSSION TITLE
This is really what I need!

### DISCUSSION THREAD

**Discussion by ubaldot** [takluyver/flit#645] [Q&A] [answered]
I am looking after a tool ONLY for packaging and publishing, whereas I only found tools that have other features that I really don't need. 
However, I have few questions

1. is it necessary to have __version__ file? Can't the version be picked directly from the pyproject.toml?
2. Is it possible to select in the pyproject.toml what to include and what to exclude in a distribution package? 

Thanks!

**Comment by takluyver** [comment, marked as answer]
> is it necessary to have version file? Can't the version be picked directly from the pyproject.toml?

Yup, you can specify `version = "0.2"` in `pyproject.toml` - if you do this, just remove version from the `dynamic` list in `pyproject.toml`. See the docs under [new style metadata](https://flit.pypa.io/en/stable/pyproject_toml.html#new-style-metadata).

> Is it possible to select in the pyproject.toml what to include and what to exclude in a distribution package?

For the sdist - the `.tar.gz` package - yes. See the docs under [Sdist section](https://flit.pypa.io/en/stable/pyproject_toml.html#sdist-section). For wheels, Flit assumes that anything inside your Python package directory (the folder containing `__init__.py`) is part of your package; this bit isn't configurable.

**Comment by ubaldot** [comment]
1. Cool and perfect because I change the version by hand. 
2. I use a src layout, so it is OK if in the wheel it is packaged everything under `./mypackage/src` (for example `./mypackage/src/mypackage, ./mypackage/src/a_sub_package, ./mypackage/src/a_bunch_of_data`) and apparently this is the case, correct?

**Reply by takluyver** [reply]
Not quite. Your package directory in that case would be `src/mypackage` (this will normally contain an `__init__.py`). You should put subpackages and data files that your code will use inside that.

There is also the possibility to include [external data](https://flit.pypa.io/en/stable/pyproject_toml.html#external-data-section), but this is less often what you want.
